src/peep/peep.c

```diff
@@ -9267,23 +9270,333 @@ static void peep_easter_egg_peep_interactions(rct_peep *peep)
 }
 
 /**
+ * rct2: 0x0069101A
+ *
+ * @return (CF)
+ */
+static bool sub_69101A(rct_map_element *esi) {
+	rct_ride *ride = get_ride(esi->properties.track.ride_index);
+	if (RCT2_ADDRESS(0x97C3AF, uint8)[ride->type] != 0) {
+		return true;
+	}
+
+	if ((uint16) ride->excitement == 0xFFFF) {
+		return false;
+	}
+
+	if (ride->excitement >= RIDE_RATING(4, 70)) {
+		return false;
+	}
+
+	if (ride->intensity >= RIDE_RATING(4, 50)) {
+		return false;
+	}
+
+	if (RideData4[ride->type].flags & 0x2000) {
+		if ((scenario_rand() & 0xFFFF) > 0x3333) {
+			return true;
+		}
+	} else if (RideData4[ride->type].flags & 0x4000) {
+		if ((scenario_rand() & 0xFFFF) > 0x1000) {
+			return true;
+		}
+	} else {
+		return true;
+	}
+
+	return false;
+}
+
+/**
  *
  *  rct2: 0x00690B99
+ *
+ * @param edge (eax)
+ * @param peep (esi)
+ * @param[out] rideToView (cl)
+ * @param[out] rideSeatToView (ch)
+ * @return !CF
  */
-static bool sub_690B99(rct_peep *peep, int edge, uint8 *rideToView, uint8 *rideSeatToView)
+static bool new_sub_690B99(rct_peep *peep, uint8 edge, uint8 *rideToView, uint8 *rideSeatToView)
 {
+	rct_map_element *mapElement, *surfaceElement;
+
+	surfaceElement = map_get_surface_element_at(peep->next_x / 32, peep->next_y / 32);
+
+	mapElement = surfaceElement;
+	do {
+		if (map_element_get_type(mapElement) != MAP_ELEMENT_TYPE_FENCE) continue;
+		if (map_element_get_direction(mapElement) != edge) continue;
+		if (g_wallSceneryEntries[mapElement->properties.fence.type]->wall.flags2 & WALL_SCENERY_FLAG4) continue;
+		if (peep->next_z + 4 <= mapElement->base_height) continue;
+		if (peep->next_z + 1 >= mapElement->clearance_height) continue;
+
+		return false;
+	} while (!map_element_is_last_for_tile(mapElement++));
+
+
+	uint16 x = peep->next_x + TileDirectionDelta[edge].x;
+	uint16 y = peep->next_y + TileDirectionDelta[edge].y;
+	if (x > 255 * 32 || y > 255 * 32) {
```

Comment:
I don't think in this case it will cause issues as there will be a map element at that address. The map element will cause the next section of code to return.